plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java

```diff
@@ -1464,4 +1448,55 @@ private String deleteDir(String dir) {
     private static String getVolumeRelativeDirInSecStroage(long volumeId) {
         return "volumes/" + volumeId;
     }
+
+    /**
+     * This method is to check if a given TaskInfo Object is valid( and has name and entity name assigned).It return true if TaskInfo Object is valid and false otherwise.
+     *
+     * @param TaskInfo
+     *        info
+     * @return boolean(true or false)
+     **/
+    private boolean isvalidTaskInfoObj(TaskInfo info){
+        return !(info == null || info.getEntityName() == null || info.getName() == null);
```

Comment:
Please check `info.getEntityName()` and `info.getName()` with StringUtils.isBlank() rather than a simple `null` check.